salt/_states/metalk8s_kubernetes.py

```diff
@@ -472,6 +472,78 @@ def namespace_present(name, **kwargs):
     return ret
 
 
+def namespace_annotation_present(
+        name,
+        annotation_key,
+        annotation_value,
+        **kwargs):
+    '''
+    Ensures that the namespace is patched.
+
+    name
+        The name of the namespace
+
+    annotation_key
+        Key of the annotation
+    annotation_value
+        Value of the annotation
+    '''
+
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    annotations = __salt__['metalk8s_kubernetes.namespace_annotations'](
+        name, **kwargs
+    ) or {}
+
+    if annotation_key not in annotations:
+        if __opts__['test']:
+            ret['comment'] = 'The annotation is going to be set'
```

Comment:
MB but you will have no `comment` if your annotation already there. You need to get this line out of the `if __opts__['test']`

Reply:
Yes, I can easily make this change.